File: DocSyncSpring/src/main/java/com/docsync/app/service/EquipmentService.java
package com.docsync.app.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.docsync.app.bean.Equipment;
import com.docsync.app.dao.EquipmentRepository;

@Service

public class EquipmentService {
	@Autowired
	private EquipmentRepository eqrepo;
	
	//add equipment
	public Equipment add(Equipment eq) {
		return eqrepo.save(eq);
	}
	
	//update by id
	public Equipment update(Long id,Equipment details) {
		Equipment equip = eqrepo.getReferenceById(id);

        equip.setName(details.getName());
        equip.setSerialNumber(details.getSerialNumber());
        equip.setStatus(details.getStatus());
        equip.setPurchaseDate(details.getPurchaseDate());

        return eqrepo.save(equip);

	}
	
	//delete by id
	public void delete(Long id) {
		  if (!eqrepo.existsById(id)) {
	            throw new RuntimeException("Equipment not found with this id: " + id);
	        }
	        eqrepo.deleteById(id);
	}
	
	// read All
    public List<Equipment> getAllEquipments() {
        return eqrepo.findAll();
    }
	
	// Method to find equipment by status
    public List<Equipment> getEquipmentByStatus(String status) {
        return eqrepo.findByStatus(status);
    }

    // Method to find equipment by serial number
    public Optional<Equipment> getEquipmentBySerialNumber(String serialNumber) {
        return eqrepo.findBySerialNumber(serialNumber);
    }
	
}

File: DocSync/src/services/equipmentService.js
import axios from '@/lib/axios';

const API_URL = "https://docsync.datavoid.in/equipment";

export const equipmentService = {
  // GET /equipment
  getAll: async () => {
    const response = await axios.get(API_URL);
    return response.data;
  },

  // POST /equipment
  create: async (equipment) => {
    const response = await axios.post(API_URL, equipment);
    return response.data;
  },

  // PUT /equipment/{id}
  update: async (id, equipment) => {
    const response = await axios.put(`${API_URL}/${id}`, equipment);
    return response.data;
  },

  // DELETE /equipment/{id}
  delete: async (id) => {
    await axios.delete(`${API_URL}/${id}`);
  },

  // GET /equipment/status/{status}
  getByStatus: async (status) => {
    const response = await axios.get(`${API_URL}/status/${status}`);
    return response.data;
  },

  // GET /equipment/serial/{serialNumber}
  getBySerial: async (serial) => {
    const response = await axios.get(`${API_URL}/serial/${serial}`);
    return response.data;
  }
};
